src/lib/derive-key-event-state.ts

```diff
@@ -0,0 +1,38 @@
+import { KeyboardEvent } from 'react'
+
+/**
+ * Returns an object with several booleans. The booleans describe the state of
+ * the `keyboardEvent` passed in as the single argument to this function. The
+ * booleans returned can be used to add custom keyboard interaction to an
+ * element within a `keydown` or `keypress` listener.
+ */
+function deriveKeyEventState(keyboardEvent: KeyboardEvent) {
+  const { key, shiftKey } = keyboardEvent
+
+  // Arrow keys
+  const isArrowDown = key === 'ArrowDown'
+  const isArrowLeft = key === 'ArrowLeft'
+  const isArrowRight = key === 'ArrowRight'
+  const isArrowUp = key === 'ArrowUp'
+
+  // Escape, Space, and Enter
+  const isEscape = key === 'Escape'
+  const isSpace = key === ' '
+
+  // Tab, Shift+Tab
+  const isShiftTab = shiftKey && key === 'Tab'
+  const isTab = !shiftKey && key === 'Tab'
+
+  return {
+    isArrowDown,
+    isArrowLeft,
+    isArrowRight,
+    isArrowUp,
+    isEscape,
+    isShiftTab,
+    isSpace,
+    isTab,
```

Comment:
TODO: update these names to include `Key` in the name. Still getting feedback on whether or not variable names for multiple keys should have `Keys` instead of `Key` (e.g. `isShiftTabKey` vs. `isShiftTabKeys`).

Reply:
Done in https://github.com/hashicorp/dev-portal/pull/241/commits/0abbdbc67e534f3e00618990272ff842fcf8e480